### PR TITLE
Patch/GUM 79 added initials to avatar

### PR DESCRIPTION
**Pull request by R-Mielamud**

### CODE REVIEW

File: src/components/avatar/avatar.tsx [hunk 1]
```diff
@@ -1,19 +1,45 @@
...
   style?: StyleProp<ViewStyle>;
 };
 
-const Avatar: React.FC<AvatarProps> = ({ size, url, style }) => {
-  if (url) {
-    return <AvatarUI.Image size={size} source={{ uri: url }} style={style} />;
+const Avatar: React.FC<AvatarProps> = ({ size, user: customUser, style }) => {
+  const { user: reduxUser } = useAppSelector(state => state.auth);
+
+  const user = useMemo(() => customUser ?? reduxUser, [customUser, reduxUser]);
```
Comment (liza-veis) [inline src/components/avatar/avatar.tsx]: I think the avatar component must be simple and not directly interact with the store

File: src/components/avatar/avatar.tsx [hunk 2]
```diff
@@ -1,19 +1,40 @@
...
-const Avatar: React.FC<AvatarProps> = ({ size, url, style }) => {
-  if (url) {
-    return <AvatarUI.Image size={size} source={{ uri: url }} style={style} />;
+const Avatar: React.FC<AvatarProps> = ({ size, user, style }) => {
+  const initials = useMemo<string>(() => {
+    if (!user) {
+      return '';
+    }
+
+    return `${user.firstName[0]}.${user.lastName[0]}.`;
```
Comment (AleksandrTermenzhy) [inline src/components/avatar/avatar.tsx]: Let's remove dots

File: src/components/avatar/avatar.tsx [hunk 3]
```diff
@@ -1,19 +1,40 @@
...
+    return (
+      <AvatarUI.Image source={{ uri: user.avatar }} size={size} style={style} />
+    );
   }
 
-  return <AvatarUI.Text size={size} label="" style={style} />;
+  return (
+    <AvatarUI.Text
+      label={initials}
+      labelStyle={{ fontSize: size / 4 }}
```
Comment (AleksandrTermenzhy) [inline src/components/avatar/avatar.tsx]: What is `4`? And why don't we just set `fontSize` explicitly?

**Review by AleksandrTermenzhy — APPROVED** [review]

File: src/components/avatar/avatar.tsx (hunk 3 above)
Comment (R-Mielamud) [reply src/components/avatar/avatar.tsx]: Because avatars can have different size.

**Review by katyakostikova — APPROVED** [review]

**Pull request merged**